<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>715</width>
    <height>497</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MakeICT Badge Printer</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>images/icon.svg</normaloff>images/icon.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="2">
    <item>
     <widget class="QTabWidget" name="previewTabs">
      <property name="tabPosition">
       <enum>QTabWidget::South</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="usesScrollButtons">
       <bool>false</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="badgePreviewTab">
       <attribute name="title">
        <string>Badge Preview</string>
       </attribute>
       <layout class="QHBoxLayout" name="verticalLayout_2" stretch="2,3">
        <item>
         <widget class="WebViewer" name="preview" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="url" stdset="0">
           <url>
            <string>about:blank</string>
           </url>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="controlsLayout">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Template</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="templateSelector"/>
            </item>
            <item row="1" column="0">
             <widget class="Line" name="line_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>QR</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLineEdit" name="qrInput">
                <property name="minimumSize">
                 <size>
                  <width>150</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>http://makeict.org/</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="testQR">
                <property name="toolTip">
                 <string>Test this QR code</string>
                </property>
                <property name="statusTip">
                 <string>Test this QR code</string>
                </property>
                <property name="whatsThis">
                 <string>This button will attempt to launch the QR code data as a URL in your default web browser.</string>
                </property>
                <property name="text">
                 <string>ðŸŒŽ</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="3" column="1">
             <widget class="Line" name="line_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Quick print</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="quickPrintSelector"/>
            </item>
            <item>
             <widget class="QToolButton" name="quickPrint">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>ðŸ–¶</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="captureButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>ðŸ“·
Capture picture</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="logButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Log only</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="waitTab">
       <attribute name="title">
        <string>Wait</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>&lt;p style=&quot;font-size: 24pt&quot;&gt;ðŸ“·&lt;/p&gt;Initializing camera...</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="cameraTab">
       <attribute name="title">
        <string>Camera</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="ClickableCameraViewfinder" name="cameraViewFinder" native="true"/>
        </item>
        <item>
         <widget class="QPushButton" name="grabCapture">
          <property name="text">
           <string>ðŸ“· Capture</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cancelCapture">
          <property name="text">
           <string>âŠ˜ Cancel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>715</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoadTemplate"/>
    <addaction name="actionSaveACopy"/>
    <addaction name="separator"/>
    <addaction name="actionCapture"/>
    <addaction name="actionImportImage"/>
    <addaction name="actionPrint"/>
    <addaction name="actionLogOnly"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionQuickPrintHelp"/>
   </widget>
   <widget class="QMenu" name="menuCameras">
    <property name="title">
     <string>&amp;Cameras</string>
    </property>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuPrint_mode">
    <property name="title">
     <string>Optio&amp;ns</string>
    </property>
    <addaction name="actionUseInkscape"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCameras"/>
   <addaction name="menuPrint_mode"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionPrint">
   <property name="text">
    <string>&amp;Print and log...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionSaveACopy">
   <property name="text">
    <string>&amp;Save a copy...</string>
   </property>
   <property name="toolTip">
    <string>SaveAcopy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCapture">
   <property name="text">
    <string>&amp;Capture image</string>
   </property>
   <property name="toolTip">
    <string>Capture or release the frame</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionLoadTemplate">
   <property name="text">
    <string>&amp;Load template...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="actionRefreshCameras">
   <property name="text">
    <string>âŸ³ &amp;Refresh</string>
   </property>
  </action>
  <action name="actionImportImage">
   <property name="text">
    <string>&amp;Import image...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionLogOnly">
   <property name="text">
    <string>Log &amp;only</string>
   </property>
  </action>
  <action name="actionUseInkscape">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use &amp;Inkscape to print</string>
   </property>
  </action>
  <action name="actionPrintImmediately">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use &amp;Print without opening dialog</string>
   </property>
  </action>
  <action name="actionQuick_print">
   <property name="text">
    <string>Quick print</string>
   </property>
  </action>
  <action name="actionUse_Inkscape_to_print">
   <property name="text">
    <string>Use Inkscape</string>
   </property>
  </action>
  <action name="actionUse_standard_print_dialog">
   <property name="text">
    <string>Use standard print dialog</string>
   </property>
  </action>
  <action name="actionQuickPrintHelp">
   <property name="text">
    <string>&amp;Quick print...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClickableCameraViewfinder</class>
   <extends>QWidget</extends>
   <header>CustomWidgets</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WebViewer</class>
   <extends>QWidget</extends>
   <header>CustomWidgets</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>templateSelector</tabstop>
  <tabstop>qrInput</tabstop>
  <tabstop>testQR</tabstop>
  <tabstop>captureButton</tabstop>
  <tabstop>previewTabs</tabstop>
  <tabstop>cancelCapture</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>captureButton</sender>
   <signal>clicked()</signal>
   <receiver>actionCapture</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>logButton</sender>
   <signal>clicked()</signal>
   <receiver>actionLogOnly</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>449</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grabCapture</sender>
   <signal>clicked()</signal>
   <receiver>actionCapture</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
